rmw_connextdds_common/src/common/rmw_config.cpp

```diff
@@ -219,3 +219,23 @@ RMW_Connext_Config::find_type_transformation(const std::string & type_transform_
     return nullptr;
   }
 }
+
+rmw_ret_t
+RMW_Connext_Config::notify_type_use(DDS_TypeCode * type_pointer)
+{
+  if (type_counter_.find(type_pointer) == type_counter_.end())
+    type_counter_[type_pointer] = 1;
+  else
+    type_counter_[type_pointer] += 1;
+
+  return RMW_RET_OK;
+}
+
+unsigned int
+RMW_Connext_Config::notify_type_disuse(DDS_TypeCode * type_pointer)
+{
+  if (type_counter_.find(type_pointer) != type_counter_.end())
+    type_counter_[type_pointer] -= 1;
+  
+  return type_counter_[type_pointer];
```

Comment:
Comments similar to other function (use braces, cache search result locally). In this case, we can also assume that this function is only ever called after the type was "registered" at least once before, so we can expect the return to not be "end()" and the count to be greater than 0.

```suggestion
  auto existing_type = network_types_.find(type_name);
  RMW_CONNEXT_ASSERT(existing_type != network_types_.end())
  RMW_CONNEXT_ASSERT(*existing_type > 0)
  *existing_type -= 1;
  return *existing_type;
```